### ISSUE TITLE
Boost::Filesystem for Windows requires wchar handling

### ISSUE THREAD

**Issue by bugmenot2** [fhanau/Efficient-Compression-Tool#6]
Hi Felix,

Can you please adjust your Boost implementation to properly support Windows paths?
~~~~cpp
main.cpp: In function 'int main(int, const char**)':
main.cpp:353:104: error: cannot convert 'const value_type* {aka const wchar_t*}' to 'const char*' for argument '1' to 'void PerFileWrapper(const char*, const ECTOptions&)'
                     for(unsigned i = 0; i < paths.size(); i++){PerFileWrapper(paths[i].c_str(), Options);}
                                                                                                        ^
main.cpp:359:65: error: cannot convert 'const value_type* {aka const wchar_t*}' to 'const char*' for argument '1' to 'void PerFileWrapper(const char*, const ECTOptions&)'
                         PerFileWrapper(paths[i].c_str(), Options);}
                                                                 ^
~~~~

As defined by Boost:
~~~~cpp
  class BOOST_FILESYSTEM_DECL path
  {
  public:

    //  value_type is the character type used by the operating system API to
    //  represent paths.

# ifdef BOOST_WINDOWS_API
    typedef wchar_t                        value_type;
    BOOST_STATIC_CONSTEXPR value_type      preferred_separator = L'\\';
# else 
    typedef char                           value_type;
    BOOST_STATIC_CONSTEXPR value_type      preferred_separator = '/';
# endif
    typedef std::basic_string<value_type>  string_type;  
    typedef std::codecvt<wchar_t, char,
                         std::mbstate_t>   codecvt_type;
~~~~

Boost also defines functions for wchar conversion in the boost::filesystem::path_traits namespace that you can use for this. Thanks!

**Comment by fhanau** [comment]
Thank you. Is it fixed with [ad36d79](https://github.com/fhanau/Efficient-Compression-Tool/commit/ad36d79f51094bae3e65e21a34145ebb0242a1f1)?

**Comment by bugmenot2** [comment]
Yup, it builds now. Thanks for the quick fix!
